pkg/dataobj/sections/internal/columnar/decoder.go

```diff
@@ -147,7 +147,7 @@ func (dec *Decoder) ReadPages(ctx context.Context, pages []*datasetmd.PageDesc)
 		ranges := make([]rangeio.Range, 0, len(pages))
 		for _, page := range pages {
 			ranges = append(ranges, rangeio.Range{
-				Data:   make([]byte, page.DataSize),
+				Data:   make([]byte, page.CompressedSize),
```

Comment:
This should be kept as DataSize. 

DataSize/DataOffset are [meant to be used together](https://github.com/grafana/loki/blob/1fdf4baa20ec307f556d0c30edf3996d0c1d70a2/pkg/dataobj/internal/metadata/datasetmd/datasetmd.proto#L177-L182) for reading the pages. 

DataSize is the size of bytes actually encoded for the page. Typically this is equal to CompressedSize, but CompressedSize is defined by the writer (rather than the encoder) and has a chance of not being accurate.